### PR TITLE
S3897: FP due to not checking IEquatable in inheritance chain

### LINKED ISSUE #1960: Fix S3897: FP due to not checking IEquatable in inheritance chain
I need to "override" `Equals(T other)`, i.e., hide base implementation with new:

```
public sealed class PersonNames : ValueObject<PersonNames>
{
	public PersonNames(string firstName, string middleName, string lastName)
	{
		FirstName = firstName;
		MiddleName = middleName;
		LastName = lastName;
	}

	public string FirstName { get; }
	public string LastName { get; }
	public string MiddleName { get; }

	public new bool Equals(PersonNames other)
	{
		return string.Equals(FirstName, other.FirstName, StringComparison.OrdinalIgnoreCase)
			&& string.Equals(MiddleName, other.MiddleName, StringComparison.OrdinalIgnoreCase)
			&& string.Equals(LastName, other.LastName, StringComparison.OrdinalIgnoreCase);
	}

	public override string ToString()
	{
		return $"{{{FirstName},{MiddleName},{LastName}}}";
	}
}

public abstract class ValueObject<T> : IEquatable<T>
        where T : ValueObject<T>
{
...

	public bool Equals(T other)
	{
		...
	}
}
```

SonarC# 7.4.0.6452 says me that I should implement `IEquatable<T>`

### PR DESCRIPTION
**Pull request by christophe-zurn-sonarsource**
Fixes #1960
Duplication is from already existing code

### CODE REVIEW

File: sonaranalyzer-dotnet/rspec/cs/S3897_c#.html [hunk 1]
```diff
@@ -5,27 +5,13 @@
 <h2>Noncompliant Code Example</h2>
 <pre>
 class MyClass  // Noncompliant
-{
-  public override bool Equals(object other)
```
Comment (andrei-epure-sonarsource) [inline sonaranalyzer-dotnet/rspec/cs/S3897_c#.html]: MyClass does not implement IEquatable. Why did you remove this example?

Comment (christophe-zurn-sonarsource) [reply sonaranalyzer-dotnet/rspec/cs/S3897_c#.html]: It follows the discussion here: https://github.com/SonarSource/sonar-dotnet/issues/420
Implementation was already updated not to raise issue in that case, only the code examples were not up-to-date
